sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsConnectionFactoryConfiguration.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.jms;
+
+import com.azure.spring.cloud.autoconfigure.jms.properties.AzureServiceBusJmsProperties;
+import org.apache.commons.pool2.PooledObject;
+import org.messaginghub.pooled.jms.JmsPoolConnectionFactory;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.jms.JmsPoolConnectionFactoryFactory;
+import org.springframework.boot.autoconfigure.jms.JmsProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jms.connection.CachingConnectionFactory;
+
+import javax.jms.ConnectionFactory;
+import java.util.stream.Collectors;
+
+/**
+ * An auto-configuration for Service Bus JMS connection factory.
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnMissingBean(ConnectionFactory.class)
+public class ServiceBusJmsConnectionFactoryConfiguration {
+
+    private static ServiceBusJmsConnectionFactory createJmsConnectionFactory(AzureServiceBusJmsProperties properties,
+                                                                             ObjectProvider<ServiceBusJmsConnectionFactoryCustomizer> factoryCustomizers) {
+        return new ServiceBusJmsConnectionFactoryFactory(properties,
+            factoryCustomizers.orderedStream().collect(Collectors.toList()))
+            .createConnectionFactory(ServiceBusJmsConnectionFactory.class);
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @ConditionalOnProperty(prefix = "spring.jms.servicebus.pool", name = "enabled", havingValue = "false",
```

Comment:
would you mind to change configuration properties of azure from `spring.*` to something like `azure.servicebus...`? It's confusing when we are setting up the spring vs other infra.

Reply:
Thanks for the suggestion @bilak, if we change this to `spring.cloud.azure.servicebus.jms.xxx`, the configuration for a JMS configuration will be like this:
```yml
spring.jms:
  cache:
    producers: false
  listener:
    acknowledgeMode: CLIENT
spring.cloud.azure.servicebus.jms:
  pricingTier: premium
  connectionString: xxxx
```
which will also be confusing since we are extending the configuration for Spring JMS with Service Bus. What's your thought on this?

Reply:
well, everything from is one vendor should be probably under that vendor namespace and then it's easier to find.
I'd suggest:
```yaml
azure:
  servicebus:
    jms: *   
```e

the same should go for other services.

Reply:
In 4.0 (our next major version) we are changing the prefix to `spring.cloud.azure.`, we will consider this suggestion.